app/controllers/searches_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SearchesController < ApplicationController
+  skip_authorization_check
+
+  def index; end
+
+  def search
+    type = params[:search][:type]
+    unchecked = params[:search][:value]
+    value = unchecked.include?('@') ? check_spelling(unchecked) : unchecked
```

Comment:
Ok, you did it. But it's the only one special symbol that should be escaped, there are some others too.

How do think, maybe, `ThinkingSphinx` has a built-in functionality to escape queries? Have you _googled_ it?